common/scala/src/main/scala/org/apache/openwhisk/http/PoolingRestClient.scala

```diff
@@ -53,9 +53,14 @@ class PoolingRestClient(
 
   //if specified, override the ClientConnection idle-timeout and keepalive socket option value
   private val timeoutSettings = {
-    ConnectionPoolSettings(system.settings.config).withUpdatedConnectionSettings { s =>
-      timeout.map(t => s.withIdleTimeout(t)).getOrElse(s)
-    }
+    val cps = ConnectionPoolSettings(system.settings.config)
+    timeout
+      .map { t =>
+        cps
```

Comment:
this doesn't NPE with the reference to `cps` that's defined on line 56?

Reply:
Not getting it ... `cps` is defined at L56 and then referred later. So should be fine?